src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -379,6 +384,169 @@ public async Task<long> SaveStateAsync(List<KeyValuePair> kvsData, CancellationT
+            {
+                expected = kvsValue.ToString();
+                actual = rcValue.ToString();
+                result = kvsValue == rcValue;
+            }
+            else if (key.StartsWith("RC@@"))
+            {
+                ReminderCompletedData kvsReminderCompletedData = this.DeserializeReminderCompletedData(key, kvsValue);
+                ReminderCompletedData rcReminderCompletedData = ReminderCompletedDataSerializer.Deserialize(rcValue);
+                expected = kvsReminderCompletedData.ToString();
+                actual = rcReminderCompletedData.ToString();
+                result = kvsReminderCompletedData.UtcTime == rcReminderCompletedData.UtcTime
+                    && kvsReminderCompletedData.LogicalTime == rcReminderCompletedData.LogicalTime;
+            }
+            else if (key.Equals("Timestamp_VLTM"))
+            {
+                LogicalTimestamp kvsLogicalTimestamp = this.DeserializeLogicalTime(key, kvsValue);
+                LogicalTimestamp rcLogicalTimestamp = LogicalTimestampSerializer.Deserialize(rcValue);
+                expected = kvsLogicalTimestamp.ToString();
+                actual = rcLogicalTimestamp.ToString();
+                result = kvsLogicalTimestamp.Timestamp == rcLogicalTimestamp.Timestamp;
+            }
+            else if (key.StartsWith("Reminder"))
+            {
+                ActorReminderData kvsActorReminderData = this.DeserializeReminder(key, kvsValue);
+                ActorReminderData rcActorReminderData = ActorReminderDataSerializer.Deserialize(rcValue);
+                expected = kvsActorReminderData.ToString();
+                actual = rcActorReminderData.ToString();
+                result = kvsActorReminderData.ActorId == rcActorReminderData.ActorId
+                    && kvsActorReminderData.Name == rcActorReminderData.Name
+                    && kvsActorReminderData.DueTime == rcActorReminderData.DueTime
+                    && kvsActorReminderData.Period == rcActorReminderData.Period
+                    && kvsActorReminderData.State == rcActorReminderData.State
+                    && kvsActorReminderData.LogicalCreationTime == rcActorReminderData.LogicalCreationTime;
+            }
+            else
+            {
+                var message = "Migration Error: Failed to parse the KVS key - " + key;
+
+                ActorTrace.Source.WriteErrorWithId(
+                    this.TraceType,
+                    this.traceId,
+                    message);
+
+                var healthInfo = new HealthInformation(this.TraceType, message, HealthState.Error);
+                healthInfo.TimeToLive = TimeSpan.MaxValue;
+                healthInfo.RemoveWhenExpired = false;
+                this.servicePartition.ReportPartitionHealth(new HealthInformation(this.TraceType, message, HealthState.Error));
+            }
+
+            if (!result)
+            {
+                ActorTrace.Source.WriteErrorWithId(
```

Comment:
Should you be throwing an exception from here, so that ActorService catches the exception and reports partition health?

The exception need not contain explicit details about what failed as that is an internal failure and user has no role to play except for reaching out to us for support.

Reply:
Done